lib/rspec/core/memoized_helpers.rb

```diff
@@ -232,6 +232,9 @@ def let(name, &block)
           # We have to pass the block directly to `define_method` to
           # allow it to use method constructs like `super` and `return`.
           raise "#let or #subject called without a block" if block.nil?
+          if method_defined?(name) && instance_method(name).owner == self
+            RSpec.warning "method :#{name} defined multiple times through let in the same context"
```

Comment:
BTW, it would be nice to add a spec for the `def foo; end; let(:foo)` case as well.  One can imagine an implementation that tracked `let` defs directly and warned on a `let` override of a prior `let` did not warn when it overrode a `def`.